flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+            'username': {'type': 'string', 'default': ''},
+            'password': {'type': 'string', 'default': ''},
+            'enable_encrypted_communication': {'type': 'boolean', 'default': False},
+            'certificates': {
+                'type': 'object',
+                'properties': {
+                    'broker_ca_cert': {'type': 'string', 'default': ''},
+                    'client_cert': {'type': 'string', 'default': ''},
+                    'client_key': {'type': 'string', 'default': ''},
+                    'validate_broker_cert': {'type': 'boolean', 'default': True},
+                    'tls_version': {'type': 'string', 'default': '', 'enum': ['tlsv1.2', 'tlsv1.1', 'tlsv1', '']},
+                },
+                'additionalProperties': False,
+            },
+            'qos': {'type': 'integer', 'minimum': 0, 'maximum': 2, 'default': 0},
+            "retain": {"type": "boolean", 'default': False},
+        },
+        'additionalProperties': False,
+        'required': ['broker_address','topic'],
+    }
+
+    def notify(self, title, message, config):
+        """
+        Publish to an MQTT topic
+        """
+        try:
+            import paho.mqtt.client as mqtt
+            from hashlib import sha256
+            import ssl
+            import time
+
+        except ImportError as e:
+            logger.verbose('Error importing paho.mqtt.client: {}', e)
+            raise DependencyError(
+                plugin_name, 'paho.mqtt.client', 'paho-mqtt python module is required for MQTT notify plugin. ImportError: %s' % e
+            )
+
+        config['title'] = title
+        config['message'] = message
+        config['payload'] = '{} - {}'.format(config.get('title'),config.get('message'))
+
+        class PublishMQTT(mqtt.Client):
+            conn_rc_description_map = { 0: 	'Connection Accepted',
+                                1: 	'Connection Refused, unacceptable protocol version - The Server does not support the level of the MQTT protocol requested by the Client',
+                                2: 	'Connection Refused, identifier rejected - The Client identifier is correct UTF-8 but not allowed by the Server',
+                                3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                                4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                                5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+            MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
+                               'MQTTv311': 	mqtt.MQTTv31 }
+
+            def __init__(self, config):
+
+                try:
+                    self.config = config
+                    self.logger = logger
+
+                    blank2None = lambda s: s or None
```

Comment:
this seems redundant, our jsonschema resolving mechanism loads the defaults under the `default` key. Also, please look at other plugins' `prepare_config()` method to see how it's done in other plugins

Reply:
@liiight - I originally tried using the schema, but I wanted to type it as string with a default of None, but I believe it complained that None was not a string in that case and threw an exception.

Is there a way to get a string but default to None?

Reply:
I'll look into the ```prepare_config()``` as suggested

Reply:
Also resolved with related comment via commit  b7881f7e458811f267deb40f9d571826fd759525